src/pybamm/plotting/quick_plot.py

```diff
@@ -479,24 +479,24 @@ def reset_axis(self):
             ):  # pragma: no cover
                 raise ValueError(f"Axis limits cannot be NaN for variables '{key}'")
 
-    def plot(self, t, dynamic=False):
+    def plot(self, t: float | list[float], dynamic: bool = False):
```

Comment:
The python 3.9 failure comes from this line, but your change is correct.

Add
```
from __future__ import annotations
```
To the very top of the file